### PR TITLE
Add mapper-extras and the RankFeatureQuery in the hlrc

### LINKED ISSUE #43634: Feature Request: Rank Feature in Java Client
The java api client does not have native support for a Rank Feature query. It seems to be the only feature missing from the specialized query section. As I need to make use of this feature would be great to see it included in the API

### PR DESCRIPTION
**Pull request by jimczi**
This change adds the support for the RankFeatureQuery in the HLRC by
providing an extra dependency on mapper-extras-client. It also removes
the dependency on lang-painless in mapper-extras which is not needed
anymore since the move of the vector field into a dedicated module.

Closes #43634

### CODE REVIEW

File: modules/mapper-extras/src/main/java/org/elasticsearch/index/query/RankFeatureQueryBuilders.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+public final class RankFeatureQueryBuilders {
+    private RankFeatureQueryBuilders() {}
+
+    /**
+     * Return a new {@link RankFeatureQueryBuilder} that will score documents as
+     * {@code boost * S / (S + pivot)} where S is the value of the static feature.
+     * @param fieldName   field that stores features
+     * @param pivot       feature value that would give a score contribution equal to weight/2, must be in (0, +Infinity)
+     * @throws IllegalArgumentException if weight is not in (0,64] or pivot is not in (0, +Infinity)
```
Comment (mayya-sharipova) [inline modules/mapper-extras/src/main/java/org/elasticsearch/index/query/RankFeatureQueryBuilders.java]: what is `weight` here from the user perspective? Should we call it `boost`?

File: docs/java-rest/high-level/query-builders.asciidoc [hunk 2]
```diff
@@ -81,6 +82,7 @@ This page lists all the available search queries with their corresponding `Query
 | {ref}/query-dsl-script-query.html[Script]                           | {query-ref}/ScriptQueryBuilder.html[ScriptQueryBuilder]                             | {query-ref}/QueryBuilders.html#scriptQuery-org.elasticsearch.script.Script-[QueryBuilders.scriptQuery()]
 | {ref}/query-dsl-percolate-query.html[Percolate]                     | {percolate-ref}/PercolateQueryBuilder.html[PercolateQueryBuilder]                   |
 | {ref}/query-dsl-wrapper-query.html[Wrapper]                         | {query-ref}/WrapperQueryBuilder.html[WrapperQueryBuilder]                           | {query-ref}/QueryBuilders.html#wrapperQuery-java.lang.String-[QueryBuilders.wrapperQuery()]
+| {ref}/query-dsl-rank-feature-query.html[Rank Feature]               | {mapper-extras-ref}/RankFeatureQuery.html[RankFeatureQueryBuilder]                  | {query-ref}/RankFeatureQueryBuilders.html#log-java.lang.String-[RankFeatureQueryBuilders.log()]
```
Comment (mayya-sharipova) [inline docs/java-rest/high-level/query-builders.asciidoc]: Is there a reason that we point only to `log` method? As we have a dedicate query builder class should we organize in a similar way as PercolateQueryBuilder without supplying the method?
```
{ref}/query-dsl-percolate-query.html[Percolate] | {percolate-ref}/PercolateQueryBuilder.html[PercolateQueryBuilder]  

{ref}/query-dsl-rank-feature-query.html[Rank Feature]  | {mapper-extras-ref}/RankFeatureQuery.html[RankFeatureQueryBuilder]


```

**Review by mayya-sharipova — APPROVED** [review]
Thanks @jimczi.  Just small documentation clarifications, otherwise LGTM

**Comment by javanna** [comment]
should this be merged?

**Pull request merged**